rex-ast/src/macros.rs

```diff
@@ -179,6 +179,18 @@ macro_rules! d {
     };
 }
 
+#[macro_export]
+macro_rules! n {
+    ($n: expr, $x:expr) => {
+        $crate::expr::Expr::Named(
+            $crate::id::Id::new(),
+            ::rex_lexer::span::Span::default(),
+            $n.to_string(),
+            Box::new($x),
+        )
+    };
+}
```

Comment:
for consistency, you need to support the other patterns too. also, be careful about expression hygiene (e.g. use `($n).to_string()` not `$n.to_string()`)

```suggestion
macro_rules! n {
    ($n: expr, $x:expr) => {
        $crate::expr::Expr::Named(
            $crate::id::Id::new(),
            ::rex_lexer::span::Span::default(),
            ($n).to_string(),
            Box::new($x),
        )
    };
    ($span:expr; $n: expr, $x:expr) => {
        $crate::expr::Expr::Named(
            $crate::id::Id::new(),
            ($span).into(),
            ($n).to_string(),
            Box::new($x),
        )
    };
    ($id:expr, $span:expr; $n: expr, $x:expr) => {
        $crate::expr::Expr::Named(
            ($id).into(),
            ($span).into(),
            ($n).to_string(),
            Box::new($x),
        )
    };
}
```